### PR TITLE
Add tests for dictionary key completion

### PR DESCRIPTION
**Pull request by maryakins**
One of the test is failing due to #455

### CODE REVIEW

File: bpython/test/test_autocomplete.py [hunk 1]
```diff
@@ -145,3 +145,38 @@ def test_formatting_takes_just_last_part(self):
...
+        result=com.matches(2,"l[" , local)
+        self.assertSetEqual(result,set())
+
+    def test_obj_that_does_not_allow_bool_tests(self):
+        com = autocomplete.DictKeyCompletion()
+        local={'mNumPy':MockNumPy(),}
+        try:
+            com.matches(7,"mNumPy[" , local)
+        except ValueError:
+            raise AssertionError("Dict key completion raised value error.")
```
Comment (sebastinas) [inline bpython/test/test_autocomplete.py]: Just let the `ValueError` pass. If a test throws anything, it is considered as failure. I'd prefer a `self.asserSetEqual(...)` as above.

**Comment by maryakins** [comment]
Also another test revealed a bug in 2.6

**Comment by thomasballinger** [comment]
Nosetests reports an error for me instead of a failure if I let the ValueError pass - do you mean we ought to consider errors failures and not worry about catching these?

**Comment by sebastinas** [comment]
Ah yes, it will be reported as error instead of failure. But as said on IRC, the expected output of that `matches` call is probably an empty set and that is what we should test for.

**Comment by thomasballinger** [comment]
Ah yeah testing for the expected behavior vs a bug definitely makes sense

**Comment by sebastinas** [comment]
Thank you, @maryakins.

**Pull request merged**